core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1552,6 +1552,20 @@ private[spark] class DAGScheduler(
     // `findMissingPartitions()` returns all partitions every time.
     stage match {
       case sms: ShuffleMapStage if stage.isIndeterminate && !sms.isAvailable =>
+        // already executed atleast once
+        if (sms.getNextAttemptId > 0) {
+          val stagesToRollback = collectSucceedingStages(sms)
+          rollBackStages(stagesToRollback)
+          // stages which cannot be rolled back were aborted which leads to removing the
+          // the dependant job(s) from the active jobs set
```

Comment:
Why don't we need to do same for the FetchFailed case?

Reply:
Here the situation is a bit different as we are in the middle of a `submitMissingTasks`. So we just end this method early as the resubmit of the missing tasks of the stage does not needed when the whole job is aborted.